### PR TITLE
Implement data flow mapper

### PR DESCRIPTION
**Pull request by mathias-baumann-sociomantic**

### CODE REVIEW

**Comment by mihails-strasuns-sociomantic** [comment]
@mathias-baumann-sociomantic this is ready to review/merge, right?

File: src/data_flow_mapper/main.d [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+    }
+
+    enum FileName = "dataflow.gv";
+    writefln("Writing graph file: %s", FileName);
+
+    auto file = File(FileName, "w");
+
+    render(file, channels, connections, apps);
+
+    /*foreach (repo, file_id; meta_info.readme_ids)
```
Comment (mihails-strasuns-sociomantic) [inline src/data_flow_mapper/main.d:204]: Left-over code

File: src/lib/json.d [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    Params:
+        path_str = string making up the path to access the field
+        obj = object to access on
+
+    Returns:
+        json object at the given path
+
+*******************************************************************************/
+
+public Json path ( string path_str ) ( Json obj )
```
Comment (mihails-strasuns-sociomantic) [inline src/lib/json.d]: Why template argument?

```D
public Json path ( Json obj, string path_str  )
{
    Json ret = obj;
    foreach (key; path_str.splitter('.'))
        ret = ret[key];
    return ret;
}
````

Comment (mathias-baumann-sociomantic) [reply src/lib/json.d]: Mind you, i just moved the function in this commit.

About the question.. I mainly thought "why not". I generally thought calculating at compile time is preferable to calculating at run time.

Comment (mathias-baumann-sociomantic) [reply src/lib/json.d]: > @mathias-baumann-sociomantic this is ready to review/merge, right?

Yes

**Comment by mathias-baumann-sociomantic** [comment]
> @mathias-baumann-sociomantic this is ready to review/merge, right?

Yes

File: src/lib/json.d (hunk 2 above)
Comment (mihails-strasuns-sociomantic) [reply src/lib/json.d]: > I generally thought calculating at compile time is preferable to calculating at run time.

Only if it is beneficial for performance or helps to improve DRY. When neither is relevant, runtime versions should always be preferred - compile faster, less binary bloat, easier to test coverage and so on.

Comment (mihails-strasuns-sociomantic) [reply src/lib/json.d]: > Mind you, i just moved the function in this commit.

I am not even pretending to review your PRs in details :P

**Comment by mihails-strasuns-sociomantic** [comment]
Would you mind merging https://github.com/sociomantic-tsunami/neptune/pull/223 first? I think this PR will be easier to resolve conflicts for as it is smaller.

**Comment by mihails-strasuns** [comment]
(needs rebase)

**Comment by mathias-baumann-sociomantic** [comment]
rebased

**Pull request merged**